### PR TITLE
configure: disable tag error bits by default

### PR DESCRIPTION
**Pull request by hzhou**
## Pull Request Description
Propagating errors in collectives is a very fragile and complex way of collective error checking and deadlock prevention. It also takes away 2 bits from the valuable tag space. Set the default to disable.

* ~this is a testing commit~ EDIT: tests are good. ready for review.
* it is noted that masking tag bits result in ucx using a slow path.

[skip warnings]

## Author Checklist
* [x] **Provide Description** 
      Particularly focus on _why_, not _what_. Reference background, issues, test failures, xfail entries, etc.
* [x] **Commits Follow Good Practice**
      Commits are self-contained and do not do two things at once. 
      Commit message is of the form: `module: short description` 
      Commit message explains what's in the commit.
* [ ] **Passes All Tests**
      Whitespace checker. Warnings test. Additional tests via comments.
* [x] **Contribution Agreement**
      For non-Argonne authors, check [contribution agreement](http://www.mpich.org/documentation/contributor-docs/). 
      If necessary, request an explicit comment from your companies PR approval manager.

### CODE REVIEW

**Comment by hzhou** [comment]
test:mpich/ch3/tcp
test:mpich/ch4/most

**Comment by hzhou** [comment]
test:mpich/ch3/tcp
test:mpich/ch4/most

**Comment by hzhou** [comment]
test:mpich/ch3/tcp
test:mpich/ch4/most

**Comment by hzhou** [comment]
test:mpich/ch3/tcp
test:mpich/ch4/most

File: src/mpid/ch4/include/mpidpre.h [hunk 1]
```diff
@@ -23,6 +23,8 @@
 #include "uthash.h"
 #include "ch4_csel_container.h"
 
+#define MPID_TAG_DEV_BITS 0
```
Comment (raffenet) [inline src/mpid/ch4/include/mpidpre.h:26]: is this correct? `ch4/ofi` utilizes tag bits for things link synchronous and huge send protocols to my recollection?

Comment (raffenet) [reply src/mpid/ch4/include/mpidpre.h:26]: I think where I am confused is when `ch4/ofi` limits the user tag bits to 20 (`MPIDI_OFI_TAG_BITS_a`), and `MPIR_Process.tag_bits` get set to reflect that, but this value never changes.

Comment (hzhou) [reply src/mpid/ch4/include/mpidpre.h:26]: They are used in `ofi_capability_sets.h`. Whether it's `_a` or `_b` depends on whether the provider supports CQ data.

Comment (raffenet) [reply src/mpid/ch4/include/mpidpre.h:26]: Long story short this define is unused so is harmless. A future PR may use it and there we can sort out how it should be used in conjunction with `MPIR_Process.tag_bits`.

**Review by raffenet — APPROVED** [review]

**Pull request merged**